ement-room.el

```diff
@@ -2847,35 +2853,35 @@ MESSAGE is the message STRING was contained in."
         :then (lambda (data)
                 (when (buffer-live-p saved-buffer)
                   (with-current-buffer saved-buffer
-                    (ement-room--insert-preview message
+                    (ement-room--insert-preview event
                                                 related-event
-                                                num
+                                                link
                                                 (progn (unless (alist-get 'og:url data)
                                                          (push (cons 'og:url link) data))
                                                        (unless (alist-get 'og:title data)
                                                          (push (cons 'og:title "[untitled page]") data))
                                                        data)))))
         :else (lambda (why) (ement-debug "Error in fetching preview: " why))))))
 
-(defun ement-room--insert-preview (message related num data)
-  "Insert link preview for DATA into MESSAGE from node RELATED if it isn't already present.
-NUM is the index for this link."
+(defun ement-room--insert-preview (message related key data)
+  "Insert link preview for HTTP response DATA into MESSAGE from node RELATED,
+if it isn't already present. KEY will be the key used for inserting."
   (cond ((alist-get 'errcode data) (ement-debug "Rate limit in url preview"))
         (data (let ((related-node (ement-room--ewoc-node-before ement-ewoc
                                                                 message
                                                                 (lambda (a b)
                                                                   (string= (ement-event-id a)
                                                                            (ement-event-id b)))
                                                                 :pred #'ement-event-p))
-                    (key (intern (concat "preview-" (number-to-string num)))))
-                (when (not (alist-get 'url-previews (ement-event-local related)))
-                  (cl-pushnew (cons 'url-previews nil) (ement-event-local related)))
-                (unless (alist-get key (map-elt (ement-event-local related) 'url-previews))
-                  (cl-pushnew (cons key data) (map-elt (ement-event-local related) 'url-previews)))
+                    (event (ement-event-local related)))
+                (unless (alist-get 'url-previews event)
+                  (cl-pushnew (cons 'url-previews nil) event))
+                (unless (alist-get key (map-elt event 'url-previews))
+                  (cl-pushnew (cons key data) (map-elt event 'url-previews)))
                 (ewoc-invalidate ement-ewoc related-node)))))
 
 (defun ement-room--format-url-preview (data)
-  "Return a pretty-printed preview for DATA."
+  "Return a pretty-printed preview for DATA, an alist of parsed JSON from the homeserver."
```

Comment:
Nitpick: To keep the first line of the docstring short, the second clause (which is just supporting information) can go in a second sentence on a second line.  (Checkdoc will complain, otherwise (although getting rid of all Checkdoc warnings is sometimes impractical)).